src/wallet/rpcwallet.cpp

```diff
@@ -2994,6 +2994,49 @@ static UniValue listwallets(const JSONRPCRequest& request)
     return obj;
 }
 
+UniValue loadwallet(const JSONRPCRequest& request)
+{
+    if (request.fHelp || request.params.size() != 1)
+        throw std::runtime_error(
+            "loadwallet \"filename\"\n"
+            "\nLoads a wallet from a wallet file or directory.\n"
+            "\nArguments:\n"
+            "1. \"filename\"    (string, required) The wallet directory or .dat file.\n"
+            "\nResult:\n"
+            "{\n"
+            "  \"name\" : <wallet_name>,        (string) The wallet name if loaded successfully.\n"
+            "}\n"
+            "\nExamples:\n"
+            + HelpExampleCli("loadwallet", "\"test.dat\"")
+            + HelpExampleRpc("loadwallet", "\"test.dat\"")
+        );
+    std::string wallet_file = request.params[0].get_str();
+    std::string error;
+
+    fs::path wallet_path = fs::absolute(wallet_file, GetWalletDir());
+    if (fs::symlink_status(wallet_path).type() == fs::file_not_found) {
+        throw JSONRPCError(RPC_WALLET_NOT_FOUND, "Wallet " + wallet_file + " not found.");
+    }
+
+    std::string dummy_warning;
+    if (!CWallet::Verify(wallet_file, false, error, dummy_warning)) {
```

Comment:
This needs some kind of lock to avoid double-loading a wallet in two rpc threads at once.

Reply:
Hmm @jnewbery this is missing https://github.com/bitcoin/bitcoin/pull/10740#discussion_r183742636.

Reply:
Re-added. Thanks @promag